python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -578,62 +585,66 @@ async def get_materialization_status_for_asset_health(self, graphene_info: Resol
                 and asset_entry.last_failed_to_materialize_storage_id is None
             ):
                 # never materialized
-                return GrapheneAssetHealthStatus.UNKNOWN
+                return GrapheneAssetHealthStatus.UNKNOWN, "No materializations"
             if asset_entry.last_failed_to_materialize_storage_id is None:
                 # last_materialization_record must be non-null, therefore the asset successfully materialized
-                return GrapheneAssetHealthStatus.HEALTHY
+                return GrapheneAssetHealthStatus.HEALTHY, None
             elif asset_entry.last_materialization_storage_id is None:
                 # last_failed_to_materialize_record must be non-null, therefore the asset failed to materialize
-                return GrapheneAssetHealthStatus.DEGRADED
+                return (
+                    GrapheneAssetHealthStatus.DEGRADED,
+                    "Failed in run",
```

Comment:
@salazarm would you rather i add the truncated run id here? the goal is to have this be
`Failed in run [abcdefgh](link to run)`

Reply:
I might need to add a resolver to get you the right run id - it won't always be the latest run for the asset since the latest run may be in progress

Reply:
Hmm, it would be better if the FE adds it.  Honestly I think maybe the frontend should control the strings and the backend just have enums for each type

Reply:
yeah i had a similar though, but figured that then the FE would need to re-calculate the numbers for the "X/Y failed" etc. maybe instead i could have like a dictionary of the value's you'd need to generate the strings?

Reply:
Yeah thats what I was thinking. Something like:

```
GrapheneHealthStatus = Union[GrapheneDegradadedFailedInRun, ...]

class GrapheneDegradaedFailedInRun 
  FailedRunId: str
```